src/integrators/sdirk_integrators.jl

```diff
@@ -725,30 +754,36 @@ end
     ndzprev = ndz
     ndz = integrator.opts.internalnorm(Δzᵧ)
     θ = ndz/ndzprev
-    if θ > 1 || ndz*(θ^(integrator.alg.max_newton_iter - iter)/(1-θ)) > κ*tol
+    if θ > 1 || ndz*(θ^(integrator.alg.max_newton_iter - iter)/(1-θ)) > κtol
       fail_convergence = true
       break
     end
     η = θ/(1-θ)
-    do_newton = (η*ndz > κ*tol)
-    @. zᵧ = zᵧ + Δzᵧ
+    do_newton = (η*ndz > κtol)
+    zᵧ .+= Δzᵧ
   end
 
   if (iter >= integrator.alg.max_newton_iter && do_newton) || fail_convergence
     integrator.force_stepfail = true
     return
   end
 
-  @. uᵧ = (uprev + d*zprev) + d*zᵧ
-
   ################################## Solve BDF2 Step
 
   ### Initial Guess From Shampine
-  a1 = (1.5 + sqrt(2)); a2 = (2.5 + 2sqrt(2)); a3 = (6 + 4.5sqrt(2))
-  @. z = a1*zprev + a2*zᵧ - a3*(uᵧ - uprev)
-  iter = 1
 
-  @. u = (uprev + ω*zprev + ω*zᵧ) + d*z
+  #=
+  Original assignments
+    @. uᵧ = (uprev + d*zprev) + d*zᵧ
+    @. z = (1.5 + sqrt(2))*zprev + (2.5 + 2sqrt(2))*zᵧ - (6 + 4.5sqrt(2))*(uᵧ - uprev)
+  with d = (2 - sqrt(2))/2 can be simplified to
+    @. z = -sqrt(2)/2*zprev + (1 + sqrt(2)/2)*zᵧ
+  =#
+  @. z = a1*zprev + a2*zᵧ
```

Comment:
You mean whether the new definition leads to the same `z` as the old one? Or whether the solution stays the same? Is there a reason to prefer the old version to the mathematically equivalent new version if there would be a slight difference (due to floating point issues maybe)?